mqtt/mqtt-broker/src/broker.rs

```diff
@@ -27,39 +28,77 @@ pub struct BrokerState {
     sessions: Vec<SessionState>,
 }
 
-pub struct Broker {
-    sender: Sender<Message>,
-    messages: Receiver<Message>,
-    sessions: HashMap<ClientId, Session>,
-    retained: HashMap<String, proto::Publication>,
+pub struct BrokerBuilder<N, Z> {
+    state: Option<BrokerState>,
+    authenticator: N,
+    authorizer: Z,
 }
 
-impl Broker {
-    pub fn new() -> Self {
-        let (sender, messages) = mpsc::channel(1024);
+impl<N, Z> BrokerBuilder<N, Z>
+where
+    N: Authenticator,
+    Z: Authorizer,
+{
+    pub fn new(authenticator: N, authorizer: Z) -> Self {
         Self {
-            sender,
-            messages,
-            sessions: HashMap::new(),
-            retained: HashMap::new(),
+            state: Option::default(),
```

Comment:
why not just `None`?

Reply:
I think default() is more convenient if some decide to use another structure instead of Option. But it could be a None here, agree.